commands/cluster-shards.md

```diff
@@ -0,0 +1,150 @@
+`CLUSTER SHARDS` returns details about the shards of the cluster.
+A shard is defined as a collection of nodes that serve the same set of slots and that replicate from each other.
+A shard may only have a single master at a given time, but may have multiple replicas.
+It is possible for a shard to not be serving any slots while still having replicas.
+
+The command is suitable to be used by Redis Cluster client libraries in order to understand the topology of the cluster.
+A client should issue this command on startup in order to retrieve the map associating cluster *hash slots* with actual node information.
+This map should be used to direct commands to the node that is likely serving the slot associated with a given command.
+In the event the command is sent to the wrong node, this command can then be used to update the topology of the cluster.
+
+The command returns an array of shards, with each shard containing two fields, 'slots' and 'nodes'. 
+
+The 'slots' field is a list of slot ranges served by this shard, stored as pair of integers storing the start and end slots of the ranges.
+For example, if a node owns the slots 1, 2, 3, 5, 7, 8, 9, the slots field would store the ranges like:
+
+```
+1) 1) "slots"
+   2) 1) (integer) 1
+      2) (integer) 3
+      3) (integer) 5
+      4) (integer) 5
+      5) (integer) 7
+      6) (integer) 9
+```
+
+The 'nodes' field contains a list of all nodes within the shard.
+Each individual node is a map of attributes that describe the node. 
+Some attributes are optional and more attributes may be added in the future. 
+The current list of attributes:
+
+* id: The unique node id for this particular node.
+* endpoint: The preferred endpoint to reach the node, see below for more information about the possible values of this field.
+* ip: The IP address to send requests to for this node.
+* hostname (optional): The announced hostname to send requests to for this node.
+* port (optional): The TCP port number of the node.
+* tls-port (optional): The TLS port of the node.
```

Comment:
Mention (here or below the list) that at least one of the ports is present.

Maybe mention that `port` (non-TLS) is present if TLS is not required.

Reply:
Tweaked wording, let me know if that seems reasonable.